stmtcacher_test.go

```diff
@@ -16,4 +17,13 @@ func TestStmtCacherPrepare(t *testing.T) {
 
 	sc.Prepare(query)
 	assert.Equal(t, 1, db.PrepareCount, "expected 1 Prepare, got %d", db.PrepareCount)
+
+	// clear statement cache
+	clearer, ok := sc.(*StmtCacher)
```

Comment:
Shouldn't need this type assertion now, right?

Reply:
Sadly as `NewStmtCacher` returns the interface it implements we do.
How about, taking your idea from earlier. We have `NewStmtCache` return the exported `StmtCache` and deprecate `NewStmtCacher`. Leaving that responding with the interface. Embrace that accept interfaces and return structs idiom.

Reply:
went ahead and gave that a go 👍